TombEngine/Game/control/box.cpp

```diff
@@ -1124,15 +1100,9 @@ int StalkBox(ItemInfo* item, ItemInfo* enemy, int boxNumber)
 		return false;
 
 	int enemyQuad = (enemy->Pose.Orientation.y / ANGLE(90.0f)) + 2;
-	
-	int boxQuad;
-	if (z > 0)
-		boxQuad = (x > 0) ? 2 : 1;
-	else
-		boxQuad = (x > 0) ? 3 : 0;
-
+	int boxQuad = z <= 0 ? (x <= 0 ? 0 : 3) : (x > 0) + 1;
 	if (enemyQuad == boxQuad)
-		return false;
+		return 0;
```

Comment:
Inconsistency. You have left some return values as bool. Change function declaration to bool, if it doesn't return anything but 0 or 1.